rust/arrow/src/compute/kernels/comparison.rs

```diff
@@ -33,56 +33,66 @@ use crate::datatypes::{ArrowNumericType, DataType};
 use crate::error::{ArrowError, Result};
 use crate::util::bit_util;
 
-/// Helper function to perform boolean lambda function on values from two arrays, this
-/// version does not attempt to use SIMD.
-macro_rules! compare_op {
-    ($left: expr, $right:expr, $op:expr) => {{
-        if $left.len() != $right.len() {
+fn infallible_compare_operation_scalar<'a,A,S,F>(left:&'a A, right: S, predicate:F) -> Result<BooleanArray>
```

Comment:
This function name is misleading, should probably rename it back to the name the macro used.